rotkehlchen/chain/ethereum/modules/yearn/graph.py

```diff
@@ -0,0 +1,424 @@
+                    }}
+                }}
+            }}
+        }}
+    }}
+    """
+)
+
+
+class YearnV2Inquirer:
+    """Reads Yearn V2 vaults information from the graph"""
+
+    def __init__(
+            self,
+            ethereum_manager: 'EthereumManager',
+            database: 'DBHandler',
+            premium: Optional[Premium],
+            msg_aggregator: MessagesAggregator,
+    ) -> None:
+        self.ethereum = ethereum_manager
+        self.database = database
+        self.msg_aggregator = msg_aggregator
+        self.premium = premium
+        self.graph = Graph('https://api.thegraph.com/subgraphs/name/salazarguille/yearn-vaults-v2-subgraph-mainnet')  # noqa: E501
+
+    def _process_deposits(self, deposits: List[Dict[str, Any]]) -> List[YearnVaultEvent]:
+        result = []
+
+        # Since multiple transactions can be made against the same token
+        # I'll save here the queried price
+        prices_cache: Dict[Asset, Price] = {}
+
+        for entry in deposits:
+            # The id returned is a composition of hash + '-' + log_index
+            _, tx_hash, log_index, _ = entry['id'].split('-')
+
+            try:
+                from_asset = EthereumToken(entry['vault']['token']['id'])
+                to_asset = EthereumToken(entry['vault']['shareToken']['id'])
+            except UnknownAsset:
+                from_str = entry['vault']['token']['symbol']
+                to_str = entry['vault']['shareToken']['symbol']
+
+                self.msg_aggregator.add_warning(
+                    f'Ignoring deposit in yearn V2 from {from_str} to '
+                    f'{to_str} because the token is not recognized.',
+                )
+                continue
+
+            # since the query of prices is expensive we check if
+            # it's in the dict and if not query it
+            from_asset_usd_price = prices_cache.get(from_asset)
+            to_asset_usd_price = prices_cache.get(to_asset)
+
+            if from_asset_usd_price is None:
+                prices_cache[from_asset] = get_usd_price_zero_if_error(
+                    asset=from_asset,
+                    time=Timestamp(int(entry['timestamp']) // 1000),
+                    location='yearn v2 vault deposit',
+                    msg_aggregator=self.msg_aggregator,
+                )
+                from_asset_usd_price = prices_cache.get(from_asset)
+
+            if to_asset_usd_price is None:
+                prices_cache[to_asset] = get_usd_price_zero_if_error(
+                    asset=to_asset,
+                    time=Timestamp(int(entry['timestamp']) // 1000),
+                    location='yearn v2 vault deposit',
+                    msg_aggregator=self.msg_aggregator,
+                )
+                to_asset_usd_price = prices_cache.get(to_asset)
+
+            assert to_asset_usd_price is not None
+            assert from_asset_usd_price is not None
+
+            from_asset_amount = token_normalized_value(
```

Comment:
in this entire function (and I guess other similar functions) it would be much better if you used the proper deserialization methods and handling of exceptions where you would ignore invalid entries with a warning and go on to the next one.

Otherwise this function has so many key errors, and potential value errors that can stop it at any point.